### DISCUSSION TITLE
Is there a recommended way to extend the functionality of the alignments track?

### DISCUSSION THREAD

**Discussion by zhangzhen** [GMOD/jbrowse-components#4169] [Q&A]
I want to programmatically extend the functionality of the alignments track, such as sorting reads according to variant patterns, viewing reads as pairs like IGV does. Is there a recommended way to do that?

Cheers,
Zhen

**Comment by cmdcolin** [comment]
hi there,
currently, this is a little tricky. you can make entirely new track/display types, but adding new e.g. sort options to the alignments track is a little tricky. in case it helps, we have some functionality kind of like that now



- view as pairs is a great feature, we do not have it but we do have "arc display" that connects paired end reads using arcs, and "read cloud" that are kinda like view as pairs, but just draws reads all over the screen according to their insert size as y-axis instead of the standard brick-type stacked layout. here is a screen recording showing a deletion using this mode. they are colored red because the 'insert size is larger than expected' with this deletion

[Screencast from 2024-01-15 17-44-34.webm](https://github.com/GMOD/jbrowse-components/assets/6511937/c360351a-6bc1-4e95-860f-019b77641956)



- sort by is also great. we do have some sort by features, I have attached a short video demonstrating. basically you can say "Sort by->Base pair" on an alignments track, and it will sort according to the base at the exact center of the screen (which you can see by going to the view menu and selecting "Show center line").

video:

[Screencast from 2024-01-15 17-39-13.webm](https://github.com/GMOD/jbrowse-components/assets/6511937/c6ca13f9-710c-4937-a21a-0d9aa55ca8e7)

hope that helps. if you wanted to add more specific features let me know, you can either modify the core codebase or i can try to advise for making a plugin!
